diff --git a/src/app/components/project/project.component.ts b/src/app/components/project/project.component.ts
index 83cea1b..e1b63a0 100644
--- a/src/app/components/project/project.component.ts
+++ b/src/app/components/project/project.component.ts
@@ -1,7 +1,9 @@
-import {Component, OnInit} from '@angular/core';
-import {CardsService} from '../../services/cards.service';
-import {CardModel} from '../../models';
-import {Observable} from 'rxjs/Observable';
+import { Component, OnInit } from '@angular/core';
+import { CardModel } from '../../models';
+import { Observable } from 'rxjs/Observable';
+import { Store, select } from '@ngrx/store';
+import { CardsState } from '../../store/cards.reducer';
+
 
 @Component({
   selector: 'app-project',
@@ -11,11 +13,14 @@ import {Observable} from 'rxjs/Observable';
 export class ProjectComponent implements OnInit {
   public cards$: Observable<CardModel[]>;
 
-  constructor(private cardsSvc: CardsService) {
-    this.cards$ = cardsSvc.cards$;
+  constructor(private store: Store<CardsState>) {
   }
 
   ngOnInit() {
+    this.cards$ = this.store
+      .pipe(
+        select('cards'),
+      );
   }
 
 }
diff --git a/src/app/store/cards.reducer.ts b/src/app/store/cards.reducer.ts
index 492cb43..91a8d17 100644
--- a/src/app/store/cards.reducer.ts
+++ b/src/app/store/cards.reducer.ts
@@ -1,5 +1,7 @@
 import {CardModel, TaskModel} from '../models';
 
+export type CardsState = CardModel[];
+
 const initialState = [
   new CardModel('Todos'),
   new CardModel('In progress'),
@@ -11,6 +13,6 @@ const initialState = [
     initialState[0].addTask(new TaskModel(title));
   });
 
-export function cardsReducer(state: any = initialState, action: any) {
+export function cardsReducer(state: CardsState = initialState, action: any): CardsState {
   return state;
 }
